File: baselib/src/main/java/com/qisan/baselib/wigget/verticaltablayout/widget/TabView.java
package com.qisan.baselib.wigget.verticaltablayout.widget;

import android.content.Context;
import android.widget.Checkable;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.TextView;

import q.rorbin.badgeview.Badge;


/**
 * @author chqiu
 *         Email:qstumn@163.com
 */
public abstract class TabView extends FrameLayout implements Checkable, ITabView {

    public TabView(Context context) {
        super(context);
    }

    @Override
    public TabView getTabView() {
        return this;
    }

    @Deprecated
    public abstract ImageView getIconView();

    public abstract TextView getTitleView();

    public abstract Badge getBadgeView();
}

File: baselib/src/main/java/com/qisan/baselib/widget/verticaltablayout/widget/TabView.kt
package com.qisan.baselib.widget.verticaltablayout.widget

import android.content.Context
import android.widget.Checkable
import android.widget.FrameLayout
import android.widget.ImageView
import android.widget.TextView
import q.rorbin.badgeview.Badge

/**
 * Created by QiSan 2022/6/18
 * package com.qisan.wanandroid.widget.verticaltablayout.widget
 */
abstract class TabView(context: Context) : FrameLayout(context), Checkable, ITabView{

    override fun getTabView(): TabView? {
        return this
    }

    @Deprecated("")
    abstract fun getIconView(): ImageView?

    abstract fun getTitleView(): TextView?

    abstract fun getBadgeView(): Badge?
}